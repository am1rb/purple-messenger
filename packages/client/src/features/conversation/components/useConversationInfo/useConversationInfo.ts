import { useParams } from "react-router-dom";

function useConversation() {
  const { username } = useParams<{ username?: string }>();
  return { username };
}

export default useConversation;
