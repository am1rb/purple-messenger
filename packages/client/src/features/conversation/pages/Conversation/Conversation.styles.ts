import { makeStyles } from "@material-ui/styles";
import { Theme } from "@material-ui/core";

export default makeStyles((theme: Theme) => ({
  root: {},
  content: { display: "flex" },
  sidePanel: {
    flexShrink: 0,
    flexGrow: 0,
    backgroundColor: "#393a66",
    width: 300,
  },
  messageZone: {
    flexShrink: 1,
    flexGrow: 1,
  },
}));
