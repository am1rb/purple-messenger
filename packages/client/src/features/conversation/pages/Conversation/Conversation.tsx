import React, { memo } from "react";
import Layout from "features/layout/components/Layout";
import ConversationList from "features/conversation/components/ConversationList";
import MessageZone from "features/message/components/MessageZone";
import useConversationInfo from "features/conversation/components/useConversationInfo";
import useStyles from "./Conversation.styles";

function Conversation() {
  const classes = useStyles();
  const { username } = useConversationInfo();

  return (
    <Layout classes={{ content: classes.content }}>
      <ConversationList
        className={classes.conversationList}
        username={username}
      />
      <MessageZone className={classes.messageZone} username={username} />
    </Layout>
  );
}

export default memo(Conversation);
