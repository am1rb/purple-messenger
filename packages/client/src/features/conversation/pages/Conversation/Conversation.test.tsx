import React from "react";
import { render } from "@testing-library/react";
import Conversation from "./Conversation";
import useConversationInfo from "features/conversation/components/useConversationInfo";

jest.mock("features/layout/components/Layout");
jest.mock("features/conversation/components/ConversationList");
jest.mock("features/message/components/MessageZone");
jest.mock("features/conversation/components/useConversationInfo");

describe("The <Conversation /> tests", () => {
  it("Should match the snapshot", () => {
    const username = "john";
    (useConversationInfo as jest.Mock).mockReturnValue({ username });
    const { container } = render(<Conversation />);
    expect(container).toMatchSnapshot();
  });
});
