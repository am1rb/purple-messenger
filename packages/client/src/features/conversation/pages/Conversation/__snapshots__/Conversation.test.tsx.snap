// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Conversation /> tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="layout"
  >
    <div
      data-testid="conversation-list"
      data-username="john"
    />
    <div
      data-testid="message-zone"
      data-username="john"
    />
  </div>
</div>
`;
